modular_skyrat/modules/modular_items/code/modular_glasses.dm

```diff
@@ -0,0 +1,297 @@
+/obj/item/clothing/glasses/hud/aviator
+  name = "aviators"
+  desc = "A pair of designer sunglasses."
+  worn_icon = 'modular_skyrat/modules/modular_items/icons/modular_glasses_mob.dmi'
+  icon_state = "aviator"
+  icon = 'modular_skyrat/modules/modular_items/icons/modular_glasses.dmi'
+  darkness_view = 1
+  flash_protect = FLASH_PROTECTION_FLASH
+  tint = 0
+  vision_correction = FALSE
+  var/toggleable = FALSE
+  var/activation_sound = 'sound/effects/pop.ogg'
+  var/off_state
+  var/on = TRUE
+
+/obj/item/clothing/glasses/hud/aviator/suicide_act(mob/living/carbon/user)
+  user.visible_message(span_suicide("[user] is posing rapidly with their [src] on... Clearly breathtaking. It looks like [user.p_theyre()] trying to commit suicide!"))
+  return OXYLOSS
+
+/obj/item/clothing/glasses/hud/aviator/fake
+  desc = "A pair of designer sunglasses. Doesn't seem like it'll block flashes."
+  flash_protect = FLASH_PROTECTION_NONE
+
+/obj/item/clothing/glasses/hud/aviator/update_icon()
+  if(on)
+    icon_state = initial(icon_state)
+  else
+    icon_state = off_state
+
+/obj/item/clothing/glasses/hud/aviator/attack_self(mob/living/user)
```

Comment:
This looks pretty similar to the attack_self code on the projectors. You might want to look into making this a proc for `/obj/item/clothing/glasses` that can be called on both the aviators and projectors. if the code does the same thing for both of them.

Reply:
adding a proc and additional to `/obj/item/clothing/glasses` would actually be modular as long as you did it inside of a Skyrat folder. It's not going to affect any already existing code because none of it uses those variables.

Reply:
you add the variables and procs inside of one of the modular skyrat files, without running into any issues. like this:
```
/obj/item/clothing/glasses
  var/mode
  var/list/modes

/obj/item/clothing/glasses/proc/swap_mode()
  (your proc goes here)
```
This is modular because it doesn't actually modify the original `/obj/item/clothing/glasses` code